File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/stage/StageFactory.java
package com.alta_v2.game.gamelogic.stage;

import com.alta_v2.game.gamelogic.data.map.MapModel;
import com.alta_v2.game.gamelogic.data.npc.NpcModel;

import java.util.List;

public interface StageFactory {

    MenuStage createMenuStage();
    MapStage createMapStage(MapModel mapModel, List<NpcModel> npcList);

}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/stage/StageFactory.kt
package com.alta_v2.game.gamelogic.stage

import com.alta_v2.game.gamelogic.data.map.MapModel
import com.alta_v2.game.gamelogic.data.npc.NpcModel

interface StageFactory {

    fun createMenuStage(): MenuStage
    fun createMapStage(mapModel: MapModel, npcList: List<NpcModel>): MapStage

}